Aquarium/Engine/Sources/Engine/Board.swift

```diff
@@ -92,6 +104,23 @@ public struct Board {
     public func rowSum(_ index: Int, _ type: Cell) -> Int {
         mat[index].reduce(0) { acc, cell in acc + (cell == type ? 1 : 0) }
     }
+    
+    public func colSum(_ index: Int, _ type: Cell) -> Int {
+        let colCells = mat.map( { row in row[index] } )
+        return colCells.reduce(0) { acc, cell in acc + (cell == type ? 1 : 0) }
+    }
+    
+    public func validCols() -> (Bool, BoardState) {
+        for i in 0 ..< size {
+            if colSum(i, .water) > colSums[i] {
+                return (false, .columnTooMuchWater(i))
+            }
+            if colSum(i, .air) > size - colSums[i] {
+                return (false, .columnTooMuchAir(i))
+            }
+        }
+        return (true, .ok)
+    }
```

Comment:
Actually might not need this method to return any `BoardState`. I'm thinking of just making it return a `Bool` if that's okay.

Reply:
BoardState will help in the long term when test cases are failing randomly on large boards, but I guess this project is small enough so sure, you can go ahead and reduce to just Bool

Reply:
@nguyenvukhang we can actually just leave both, no harm done anyways.